pkg/build/strategies/sti/sti.go

```diff
@@ -642,7 +642,7 @@ func (builder *STI) Execute(command string, user string, config *api.Config) err
 	}
 	// Do not wait for source input if the container times out.
 	// FIXME: this potentially leaks a goroutine.
-	if !util.IsTimeoutError(err) {
+	if err != nil {
```

Comment:
@rhcarvalho that's my bad
It should really read if err == nil then wait... otherwise return the error
Right code now pushed